flytekit/core/data_persistence.py

```diff
@@ -0,0 +1,382 @@
+            return path.replace("file://", "", 1)
+        return path
+
+    def listdir(self, path: str, recursive: bool = False) -> typing.Generator[str, None, None]:
+        if not recursive:
+            files = os.listdir(self.strip_file_header(path))
+            for f in files:
+                yield f
+            return
+
+        for root, subdirs, files in os.walk(self.strip_file_header(path)):
+            for f in files:
+                yield os.path.join(root, f)
+        return
+
+    def exists(self, path: str):
+        return os.path.exists(self.strip_file_header(path))
+
+    def get(self, from_path: str, to_path: str, recursive: bool = False):
+        if from_path != to_path:
+            if recursive:
+                _dir_util.copy_tree(self.strip_file_header(from_path), self.strip_file_header(to_path))
+            else:
+                copyfile(self.strip_file_header(from_path), self.strip_file_header(to_path))
+
+    def put(self, from_path: str, to_path: str, recursive: bool = False):
+        if from_path != to_path:
+            if recursive:
+                _dir_util.copy_tree(self.strip_file_header(from_path), self.strip_file_header(to_path))
+            else:
+                # Emulate s3's flat storage by automatically creating directory path
+                self._make_local_path(os.path.dirname(self.strip_file_header(to_path)))
+                # Write the object to a local file in the sandbox
+                copyfile(self.strip_file_header(from_path), self.strip_file_header(to_path))
+
+    def construct_path(self, _: bool, add_prefix: bool, *args) -> str:
+        # Ignore add_protocol for now. Only complicates things
+        if add_prefix:
+            return os.path.join(self.default_prefix, *args)
+        return os.path.join(*args)
+
+
+class FileAccessProvider(object):
+    """
+    This is the class that is available through the FlyteContext and can be used for persisting data to the remote
+    durable store.
+    """
+
+    def __init__(self, local_sandbox_dir: Union[str, os.PathLike], raw_output_prefix: str):
+        # Local access
+        if local_sandbox_dir is None or local_sandbox_dir == "":
+            raise Exception("Can't use empty path")
+        local_sandbox_dir_appended = os.path.join(local_sandbox_dir, "local_flytekit")
+        self._local_sandbox_dir = pathlib.Path(local_sandbox_dir_appended)
+        self._local_sandbox_dir.mkdir(parents=True, exist_ok=True)
+        self._local = DiskPersistence(default_prefix=local_sandbox_dir_appended)
+
+        self._default_remote = DataPersistencePlugins.find_plugin(raw_output_prefix)(default_prefix=raw_output_prefix)
+        self._raw_output_prefix = raw_output_prefix
+
+    @staticmethod
+    def is_remote(path: Union[str, os.PathLike]) -> bool:
+        """
+        Deprecated. Lets find a replacement
+        """
+        if path.startswith("/") or path.startswith("file://"):
+            return False
+        return True
```

Comment:
nit - conciseness: feel free to ignore

```suggestion
        return not (path.startswith("/") or path.startswith("file://"))
```

Reply:
sure